lib/eth/abi/event.rb

```diff
@@ -24,20 +24,97 @@ module Abi
     module Event
       extend self
 
+      # Compute topic for ABI event interface.
+      #
+      # @param interface [Hash] ABI event interface.
+      # @return [String] a hex-string topic.
+      def compute_topic(interface)
+        sig = Abi.signature(interface)
+        Util.prefix_hex(Util.bin_to_hex(Util.keccak256(sig)))
+      end
+
+      # A decoded event log.
+      class LogDescription
+        # The event ABI interface used to decode the log.
+        attr_accessor :event_interface
+
+        # The the input argument of the event.
+        attr_accessor :args
+
+        # The named input argument of the event.
+        attr_accessor :kwargs
+
+        # The topic hash.
+        attr_accessor :topic
+
+        # Decodes event log argument values.
+        #
+        # @param event_interface [Hash] event ABI type.
+        # @param log [Hash] transaction receipt log
+        def initialize(event_interface, log)
+          @event_interface = event_interface
+
+          inputs = event_interface.fetch("inputs")
+          data = log.fetch("data")
+          topics = log.fetch("topics", [])
+          anonymous = event_interface.fetch("anonymous", false)
+
+          @topic = topics[0] if !anonymous
```

Comment:
This is a false condition, so we want to use unless.
```suggestion
          @topic = topics[0] unless anonymous
```